es-space-mgmt/main/default/classes/reservationManagerController.cls

```diff
@@ -5,54 +5,50 @@ public with sharing class reservationManagerController {
-                        'New - Not Contacted',
-                        'New'
-                    )
-                WITH SECURITY_ENFORCED
-                LIMIT 5
-            ]) {
-                customerServices.Customer c = new customerServices.Customer(
-                    l.Name,
-                    l.Email,
-                    l.City,
-                    l.State,
-                    l.Status,
-                    l.Id
-                );
-                customers.add(c);
-            }
+        Customer_Fields__mdt c = [
+            SELECT
+                Customer_Name__r.QualifiedAPIName,
+                Customer_Email__r.QualifiedAPIName,
+                Customer_City__r.QualifiedAPIName,
+                Customer_State__r.QualifiedAPIName,
+                Customer_Status__r.QualifiedAPIName,
+                Customer_Draft_Status_Values__c
+            FROM Customer_Fields__mdt
+            WHERE Sobject_Type__r.QualifiedAPIName = :sObjectType
+            WITH SECURITY_ENFORCED
+            LIMIT 1
+        ];
+
+        List<String> draftStatus = new List<String>();
+        draftStatus = c.Customer_Draft_Status_Values__c.split(',');
+
+        String query = 'SELECT ';
+        query = query + c.Customer_Name__r.QualifiedAPIName + ', ';
+        query = query + c.Customer_Email__r.QualifiedAPIName + ', ';
+        query = query + c.Customer_City__r.QualifiedAPIName + ', ';
+        query = query + c.Customer_State__r.QualifiedAPIName + ', ';
+        query = query + c.Customer_Status__r.QualifiedAPIName + ', ';
+        query = query + 'Id';
+        query = query + ' FROM ' + sObjectType;
+        if (draftStatus.size() > 0) {
+            query =
+                query +
+                ' WHERE ' +
+                c.Customer_Status__r.QualifiedAPIName +
+                ' IN: draftStatus';
         }
-        if (sObjectType == 'Contact') {
-            for (Contact c : [
-                SELECT
-                    Id,
-                    Email,
-                    Name,
-                    MailingCity,
-                    MailingState,
-                    Reservation_Status__c
-                FROM Contact
-                WHERE Reservation_Status__c IN ('Not Started')
-                WITH SECURITY_ENFORCED
-                LIMIT 5
-            ]) {
-                customerServices.Customer cu = new customerServices.Customer(
-                    c.Name,
-                    c.Email,
-                    c.MailingCity,
-                    c.MailingState,
-                    c.Reservation_Status__c,
-                    c.Id
-                );
-                customers.add(cu);
-            }
+        query = query + ' WITH SECURITY_ENFORCED LIMIT 5';
+
+        for (Sobject sobj : Database.query(query)) {
+            customerServices.Customer customer = new customerServices.Customer(
+                (String) sobj.get(c.Customer_Name__r.QualifiedAPIName),
```

Comment:
Shouldn't we catch/throw an exception on these lines if the fields do no exist or if we do not have access to them?

Reply:
@pozil The error handling is missing everywhere in this application! I can take a stab at it for this class and may be we handle for rest as we touch base the other parts of the app.

Reply:
Ok fair enough, don't bother then.